app/domain/util/concurrency_limited_cache.rb

```diff
@@ -41,14 +42,15 @@ def call(**args)
 
       @semaphore.synchronize do
         recalculate(args)
-        @cache[args]
+        @cache[cache_key]
       end
     end
 
     private
 
     def recalculate(args)
```

Comment:
why it is not aligned with the other cache class? 
` def recalculate(args, cache_key)`